src/app/shared/validators/password.validator.ts

```diff
@@ -0,0 +1,16 @@
+import { Validators, FormControl, AbstractControl } from '@angular/forms';
+
+/**
+ * @param AbstractControl which in this case is formGroup only 
+ * FormGroup extends from AbstractControl afterall.
+ * Using FormGroup will also work here.
+ */
+export function passwordValidator(c: AbstractControl): {[key: string]:boolean }|any  {
+    const password = c.get('password');
+    const password_confirmation = c.get('password_confirmation');
+    // Don't raise nomatch for empty case, required should handle it
+    if( !password.value || !password_confirmation.value) {
```

Comment:
you can use `touched` in html tag itself for checking if the input was clicked or not...
instead of checking for null case here.
Another thing can be done by disabling the submit button until every input value is filled.
 Something like this...
https://github.com/tamasfoldi/ngrx-auth/blob/master/src/app/components/login/login.component.html